builder/proxmox/common/config.go

```diff
@@ -395,8 +415,31 @@ func (c *Config) Prepare(upper interface{}, raws ...interface{}) ([]string, []st
 		}
 	}
 
+	for _, device := range c.PCIDevices {
+		if device.Host == "" && device.Mapping == "" {
+			errs = packersdk.MultiErrorAppend(errs, errors.New("either the host or the mapping key must be specified"))
+		}
+		if device.LegacyIGD != nil && *device.LegacyIGD {
+			if c.Machine != "pc" && !strings.HasPrefix(c.Machine, "pc-i440fx") {
+				errs = packersdk.MultiErrorAppend(errs, errors.New("legacy_igd requires pc-i440fx machine type"))
+			}
+			if c.VGA.Type != "none" {
+				errs = packersdk.MultiErrorAppend(errs, errors.New("legacy_igd requires vga.type set to none"))
+			}
+		}
+		if device.PCIe != nil && *device.PCIe {
+			if c.Machine != "q35" && !strings.HasPrefix(c.Machine, "pc-q35") {
+				errs = packersdk.MultiErrorAppend(errs, errors.New("pcie requires q35 machine type"))
+			}
+		}
+	}
+
 	if errs != nil && len(errs.Errors) > 0 {
 		return nil, warnings, errs
 	}
 	return nil, warnings, nil
 }
+
+func boolPtr(v bool) *bool {
```

Comment:
Since `Trilean` exists, I believe this can be removed, unless there's something that `Trilean` doesn't support and you think it's still relevant? Always open to discuss :)